rollouts/controllers/rollout_controller.go

```diff
@@ -294,26 +382,24 @@ func (r *RolloutReconciler) computeTargets(ctx context.Context,
 	return targets, nil
 }
 
-func (r *RolloutReconciler) rolloutTargets(ctx context.Context, rollout *gitopsv1alpha1.Rollout, targets *Targets) ([]gitopsv1alpha1.ClusterStatus, error) {
+func (r *RolloutReconciler) rolloutTargets(ctx context.Context, rollout *gitopsv1alpha1.Rollout, wave *gitopsv1alpha1.Wave, targets *Targets, waveAlreadyInProgress bool) (bool, []gitopsv1alpha1.ClusterStatus, error) {
 	clusterStatuses := []gitopsv1alpha1.ClusterStatus{}
 
-	if rollout.Spec.Strategy.Type != gitopsv1alpha1.AllAtOnce && rollout.Spec.Strategy.Type != gitopsv1alpha1.RollingUpdate {
-		return clusterStatuses, fmt.Errorf("%v strategy not supported yet", rollout.Spec.Strategy.Type)
-	}
-
 	concurrentUpdates := 0
-	maxConcurrent := math.MaxInt
+	maxConcurrent := int(wave.MaxConcurrent)
+	waiting := "Waiting"
+
+	if waveAlreadyInProgress {
```

Comment:
I think you will have to walk me through it.

Reply:
Completed offline.